src/entryPoints/AuthModule/TotpAuth/TotpSetup.res

```diff
@@ -2,104 +2,97 @@ let h2TextStyle = HSwitchUtils.getTextClass((H2, Optional))
 let p2Regular = HSwitchUtils.getTextClass((P2, Regular))
 let p3Regular = HSwitchUtils.getTextClass((P3, Regular))
 
-module ConfirmPopUpElement = {
+module EnterAccessCode = {
   @react.component
-  let make = (~recoveryCodes, ~downloadRecoveryCodes) => {
-    open LogicUtils
+  let make = (~setTotpStatus, ~onClickVerifyAccessCode) => {
     let showToast = ToastState.useShowToast()
+    let verifyRecoveryCodeLogic = TotpHooks.useVerifyRecoveryCode()
+    let (recoveryCode, setRecoveryCode) = React.useState(_ => "")
+    let (buttonState, setButtonState) = React.useState(_ => Button.Normal)
 
-    let firstHalf = recoveryCodes->Array.copy
-    firstHalf->Array.splice(~start=0, ~remove=recoveryCodes->Array.length / 2, ~insert=[])
+    let verifyAccessCode = async () => {
+      try {
+        open LogicUtils
 
-    let secondHalf = recoveryCodes->Array.copy
-    secondHalf->Array.splice(
-      ~start=recoveryCodes->Array.length / 2,
-      ~remove=recoveryCodes->Array.length,
-      ~insert=[],
-    )
+        setButtonState(_ => Button.Loading)
 
-    let onCopyClick = ev => {
-      ev->ReactEvent.Mouse.stopPropagation
-      Clipboard.writeText(JSON.stringifyWithIndent(recoveryCodes->getJsonFromArrayOfString, 3))
-      showToast(~message="Copied to Clipboard!", ~toastType=ToastSuccess, ())
+        if recoveryCode->String.length > 0 {
+          let body = [("recovery_code", recoveryCode->JSON.Encode.string)]->getJsonFromArrayOfJson
+          let _ = await verifyRecoveryCodeLogic(body)
+          onClickVerifyAccessCode(false)->ignore
+        } else {
+          showToast(~message="OTP field cannot be empty!", ~toastType=ToastError, ())
```

Comment:
why error message is telling `OTP field cannot be empty!` it should be `Access Code  cannot be empty`